bench.py

```diff
@@ -9,42 +9,99 @@
 
 root_dir = os.path.dirname(os.path.realpath(__file__))
 
-code_filename = os.path.join(root_dir, 'input', 'xul.dll')
-file_code_offs = 0x400
-file_code_len = 0x2460400
-code_loop_count = 20
-
-if len(sys.argv) == 1:
-    pass
-elif len(sys.argv) == 4 or len(sys.argv) == 5:
-    file_code_offs = int(sys.argv[1])
-    file_code_len = int(sys.argv[2])
-    code_filename = sys.argv[3]
-    if len(sys.argv) >= 5:
-        code_loop_count = int(sys.argv[4])
-    else:
-        # Match the old file + loop-count
-        code_loop_count = round(0x2460400 * 20 / file_code_len)
-else:
-    print('Expected no args or:')
-    print(f'  {sys.argv[0]} <code-offset> <code-len> <filename> [loop-count]')
+x86_bench_defaults = {
+        "arch": "x86",
+        "filename": os.path.join(root_dir, 'input', 'xul.dll'),
+        "file_code_offs": 0x400,
+        "file_code_len": 0x2460400,
+        "code_loop_count": 20,
+}
+
+armv7_bench_defaults = {
+        "arch": "armv7",
+        "filename": os.path.join(root_dir, 'input', 'tdfx_dri.so'),
```

Comment:
i suppose i didn't commit it.. i stole a reasonably chunky .so out of x11 after crawling through debian armv7 packages. i was hoping to find an armv7 version of `xul.dll`, but no one builds firefox for aarch32 anymore... but there's probably a more directly comparable object for armv8 at least :3

Reply:
Debian seems to have ARM32 builds of Firefox as well:

http://ftp.us.debian.org/debian/pool/main/f/firefox/firefox_144.0-1_armhf.deb

Though it's also fine with me to use that other lib that you referred to above.